Assets/__Scripts/MapEditor/Grid/GridRenderingController.cs

```diff
@@ -78,13 +80,21 @@ private void GridMeasureSnappingChanged(int snapping)
             g.SetPropertyBlock(preciseBeatPropertyBlock);
         }
 
-        UpdateHighContrastGrids();
+        UpdateGridColors();
     }
 
-    private void UpdateHighContrastGrids(object _ = null)
+    private void UpdateGridColors(object _ = null)
     {
-        highContrastBeatPropertyBlock.SetFloat(MainAlpha, Settings.Instance.HighContrastGrids ? 0 : MainAlphaDefault);
-        foreach (Renderer g in gridsToDisableForHighContrast) g.SetPropertyBlock(highContrastBeatPropertyBlock);
+        float gridAlpha = Settings.Instance.GridTransparency;
+        Color newColor = Settings.Instance.HighContrastGrids ? MainColorHighContrast : MainColorDefault;
+        newColor.a = 1f - gridAlpha;
+        Shader shaderToUse =  Shader.Find(newColor.a == 1f ? "Unlit/Color" : "Unlit/Transparent Colored");
+        beatColorPropertyBlock.SetColor(MainColor, newColor);
+        foreach (Renderer g in gridsToColor)
+        {
+            g.material.shader = shaderToUse;
```

Comment:
accessing `g.material` will automatically create a new material instance and essentially defeat the whole point of using MaterialPropertyBlocks (saving render calls by re-using the *exact* same material)

I'd recommend finding a different approach that doesnt involve shader switching.

Reply:
Would creating two base objects and showing/hiding them depending on what shader is needed be too jank?

Reply:
not the jankiest thing i've ever done